lib/log4js.js

```diff
@@ -427,17 +427,19 @@ function requireAppender(appender) {
  * @private
  */
 function loadAppender(appender, appenderModule) {
-  appenderModule = appenderModule || requireAppender(appender);
+  if (!module.exports.appenders[appender]) {
```

Comment:
Does this code work? When you call `log4js.configure` it calls `clearAppenders`, which resets this appenders object, so it should be empty when reading your config file.